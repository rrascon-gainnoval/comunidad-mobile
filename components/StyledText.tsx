import React, { useContext } from "react";
import { Text, TextProps } from "./Themed";
import { StyleSheet } from "react-native";
import { theme } from "../constants/Theme";
import { ThemeContext } from "../Theme.Provider";

export function MonoText(props: TextProps) {
  return (
    <Text {...props} style={[props.style, { fontFamily: "space-mono" }]} />
  );
}

export const HeaderText = (props: TextProps) => {
  const { primaryColor } = useContext(ThemeContext);

  return (
    <Text
      style={[
        styles.headerText,
        { color: primaryColor },
        styles.margins,
        props.style,
      ]}
    >
      {props.children}
    </Text>
  );
};

export const TitleText = (props: TextProps) => {
  return (
    <Text style={[styles.titleText, styles.margins, props.style]}>
      {props.children}
    </Text>
  );
};

const styles = StyleSheet.create({
  headerText: {
    fontSize: 28,
    fontWeight: "bold",
    textAlign: "center",
  },
  titleText: {
    fontSize: 20,
    fontWeight: "bold",
  },
  margins: {
    marginHorizontal: theme.marginX,
    marginVertical: theme.marginY,
  },
});
