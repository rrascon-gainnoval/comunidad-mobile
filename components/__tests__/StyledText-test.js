import * as React from "react";
import renderer from "react-test-renderer";

import { MonoText, HeaderText } from "../StyledText";

describe("<MonoText/>", () => {
  it("Renders correctly", () => {
    const tree = renderer.create(<MonoText>Snapshot test!</MonoText>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<HeaderText/>", () => {
  it("Renders correctly", () => {
    const tree = renderer
      .create(<HeaderText>Snapshot test!</HeaderText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
