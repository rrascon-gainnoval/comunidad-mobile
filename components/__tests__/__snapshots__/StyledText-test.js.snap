// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderText/> Renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#575757",
      },
      Array [
        Object {
          "color": "#42D2B0",
          "fontSize": 28,
          "fontWeight": "bold",
        },
        Object {
          "marginHorizontal": 10,
          "marginVertical": 10,
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`<MonoText/> Renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#575757",
      },
      Array [
        undefined,
        Object {
          "fontFamily": "space-mono",
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;
